pymc3/tests/test_step.py

```diff
@@ -238,3 +241,129 @@ def test_binomial(self):
             Binomial('x', 10, 0.5)
             steps = assign_step_methods(model, [])
         self.assertIsInstance(steps, Metropolis)
+
+
+class TestSampleEstimates(SeededTest):
+    def test_parameter_estimate(self):
+        alpha_true, sigma_true = 1, 0.5
+        beta_true = np.array([1, 2.5])
+
+        size = 100
+
+        X1 = np.random.randn(size)
+        X2 = np.random.randn(size) * 0.2
+        Y = alpha_true + beta_true[0] * X1 + beta_true[1] * X2 + np.random.randn(size) * sigma_true
+
+        with Model() as model:
+            alpha = Normal('alpha', mu=0, sd=10)
+            beta = Normal('beta', mu=0, sd=10, shape=2)
+            sigma = Uniform('sigma', lower=0.0, upper=1.0)
+            mu = alpha + beta[0] * X1 + beta[1] * X2
+            Y_obs = Normal('Y_obs', mu=mu, sd=sigma, observed=Y)
+
+            for step_method in (NUTS(), Metropolis(),
```

Comment:
I can put them back together if you want. I felt it might be good to have them broken up, so that we can understand what breaking changes do what. Metropolis is different to NUTS.